curvefs/test/mds/mds_service_test.cpp

```diff
@@ -128,6 +147,155 @@ class MdsServiceTest : public ::testing::Test {
             fsStorage_, spaceManager_, metaserverClient_, topoManager_,
             s3Adapter_, nullptr, fsManagerOption);
         ASSERT_TRUE(fsManager_->Init());
+
+        mdsService_ = std::make_shared<MdsServiceImpl>(fsManager_, nullptr);
+        // add metaserver service
+        ASSERT_EQ(server.AddService(mdsService_.get(),
+                  brpc::SERVER_DOESNT_OWN_SERVICE), 0);
+
+        ASSERT_EQ(server.AddService(&metaserverService,
+                                    brpc::SERVER_DOESNT_OWN_SERVICE), 0);
+
+        ASSERT_EQ(
+            server.AddService(&mockCliService2,
+                              brpc::SERVER_DOESNT_OWN_SERVICE), 0);
+
+        ASSERT_EQ(0, server.AddService(&fakeCurveFSService,
+                                    brpc::SERVER_DOESNT_OWN_SERVICE));
+
+        // start rpc server
+        ASSERT_EQ(server.Start(kRpcServerAddr.c_str(), &option), 0);
+
+        // init client
+        ASSERT_EQ(channel.Init(server.listen_address(), nullptr), 0);
+        stub_ = new MdsService_Stub(&channel);
```

Comment:
since you have mentioned smared pointers are modern way of managing pointers, there should be no more raw `new/delete`.

Reply:
modified